yb-voyager/src/anon/sql_anonymizer.go

```diff
@@ -396,111 +480,794 @@ func (a *SqlAnonymizer) miscellaneousNodesProcessor(msg protoreflect.Message) (e
+			str := rs.Object.GetString_()
+			if str.Sval != "" {
+				var err error
+				str.Sval, err = a.registry.GetHash(prefix, str.Sval)
+				if err != nil {
+					return fmt.Errorf("anon rename object: %w", err)
+				}
+			}
+		}
+	}
+
+	// Always anonymize the new name
+	if rs.Newname != "" {
+		var err error
+		rs.Newname, err = a.registry.GetHash(prefix, rs.Newname)
 		if err != nil {
-			return fmt.Errorf("anon ColumnRef, fieldLen=4, table field: %w", err)
+			return fmt.Errorf("anon rename newname: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// handleGenericDropStmt processes DROP statements for any object type
+func (a *SqlAnonymizer) handleGenericDropStmt(ds *pg_query.DropStmt) error {
+	if ds == nil || len(ds.Objects) == 0 {
+		return nil
+	}
+
+	prefix := a.getObjectTypePrefix(ds.RemoveType)
+
+	// Process each object in the DROP statement
+	for _, obj := range ds.Objects {
+		var err error
+
+		// Handle different object name patterns based on object type
+		switch ds.RemoveType {
+		case pg_query.ObjectType_OBJECT_SCHEMA:
+			// Schema names are simple strings
+			if str := obj.GetString_(); str != nil && str.Sval != "" {
+				str.Sval, err = a.registry.GetHash(prefix, str.Sval)
+			}
+
+		case pg_query.ObjectType_OBJECT_DOMAIN:
+			// Domain names can be in TypeName format
+			if tn := obj.GetTypeName(); tn != nil {
+				err = a.anonymizeStringNodes(tn.Names, prefix)
+			}
+
+		default:
+			// For all other object types, try as qualified name list first, then as simple string
+			if list := obj.GetList(); list != nil {
+				// Use existing anonymizeStringNodes for qualified names
+				err = a.anonymizeStringNodes(list.Items, prefix)
+			} else if str := obj.GetString_(); str != nil && str.Sval != "" {
+				// Fallback to simple string
+				str.Sval, err = a.registry.GetHash(prefix, str.Sval)
+			}
 		}
-		col.Sval, err = a.registry.GetHash(COLUMN_KIND_PREFIX, col.Sval)
+
 		if err != nil {
-			return fmt.Errorf("anon ColumnRef, fieldLen=4, column field: %w", err)
+			return fmt.Errorf("anon drop object: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// handleGenericAlterObjectSchemaStmt processes ALTER ... SET SCHEMA statements for TABLE, EXTENSION, TYPE, and SEQUENCE object types
+func (a *SqlAnonymizer) handleGenericAlterObjectSchemaStmt(aos *pg_query.AlterObjectSchemaStmt) error {
+	if aos == nil {
+		return nil
+	}
+
+	// Handle TABLE, EXTENSION, TYPE, and SEQUENCE
+	switch aos.ObjectType {
+	case pg_query.ObjectType_OBJECT_TABLE:
+		// Tables use RangeVar for their names
+		if aos.Relation != nil {
+			err := a.anonymizeRangeVarNode(aos.Relation, TABLE_KIND_PREFIX)
+			if err != nil {
+				return fmt.Errorf("anon alter table schema: %w", err)
+			}
+		}
+
+	case pg_query.ObjectType_OBJECT_SEQUENCE:
+		// Sequences use RangeVar for their names
+		if aos.Relation != nil {
+			err := a.anonymizeRangeVarNode(aos.Relation, SEQUENCE_KIND_PREFIX)
+			if err != nil {
+				return fmt.Errorf("anon alter sequence schema: %w", err)
+			}
```

Comment:
can combine these two as anonymization is same just the prefix is different which you can get from the helper function `getrObjectTypePrefix` I belive